galaxy_importer/ansible_test/builders/pulp_build.py

```diff
@@ -15,7 +15,271 @@
+        r = requests.post(
+            url=f'{self.api_url}/pulp/api/v3/repositories/container/container/',
+            auth=self.basic_auth,
+            json={'name': f'ansible_test_repo_{str(uuid.uuid4())}'}
+        )
+        if r.status_code != 201:
+            raise exceptions.AnsibleTestError(
+                f'Could not create ContainerRepository: \
+                    {r.status_code} {r.reason} {r.text}')
+        return r.json()['pulp_href']
+
+    def _get_container_distribution_href(self):
+        pulp_url = '/pulp/api/v3/distributions/container/container/'
+        r = requests.post(
+            url=f'{self.api_url}{pulp_url}',
+            auth=self.basic_auth,
+            json={
+                'name': self.registry_name,
+                'base_path': self.registry_base_path,
+                'repository': self.container_repository_href
+            }
+        )
+        if r.status_code == 202:
+            return self._wait_for_dist_build(r.json()['task'])
+        else:
+            raise exceptions.AnsibleTestError(
+                f'Could not create ContainerDistribution: \
+                    {r.status_code} {r.reason} {r.text}')
+
+    def _wait_for_dist_build(self, task_href):
+        for i in range(API_CHECK_RETRIES):
+            r = requests.get(
+                url=f'{self.api_url}{task_href}',
+                auth=self.basic_auth)
+            status = r.json()['state']
+            if status == 'failed' or status == 'canceled':
+                self.cleanup()
+                raise exceptions.AnsibleTestError(
+                    f'Could not create ContainerDistribution: \
+                        {r.status_code} {r.reason} {r.text}')
+            elif status == 'completed':
+                return r.json()['created_resources'][0]
+            time.sleep(API_CHECK_DELAY_SECONDS)
+
+    def _get_registry_href(self):
+        r = requests.get(
+            url=f'{self.api_url}{self.container_distribution_href}',
+            auth=self.basic_auth,
+        )
+        if r.status_code == 200:
+            self._add_registry_to_conf(r.json()['registry_path'])
+            return r.json()['registry_path']
+        else:
+            raise exceptions.AnsibleTestError(
+                f'Could not retrieve Registry href: \
+                    {r.status_code} {r.reason} {r.text}')
+
+    def _add_registry_to_conf(self, registry_href):
+        reg_file = f'/home/{os.getlogin()}/.config/containers/registries.conf'
+        if not os.path.exists(reg_file):
+            path = f'home/{os.getlogin()}/.config/containers'.split('/')
+            for index, item in enumerate(path):
+                current_path = Build._get_current_path(index, path)
+                if not os.path.exists(current_path):
+                    subprocess.run(['mkdir', f'{current_path}'])
+                else:
+                    continue
+            subprocess.run(['touch', f'{reg_file}'])
+
+        with open(reg_file, 'r+') as f:
```

Comment:
when are the cases the `reg_file` exists with text versus not? I'm wondering if there a way to make it more clear which code path is being followed and why